### PR TITLE
VAULT-6727 Add role resolution to OCI

### PR DESCRIPTION
**Pull request by VioletHynes**
Support role resolution functionality (to work with role based quotas). The intent of this operation is to take the same data as a login request and return the role that would be used in the request, so that we can quota upon login requests that have a certain role.

Note that a new path was made, /login, as OCI usually uses /login/role. Core checks the /login endpoint for role resolution, and this endpoint must be generic as we need to find the login request without necessarily knowing the backend.

Afterwards, I will push a new tag and pull this new version into Vault.

### CODE REVIEW

File: path_login.go [hunk 1]
```diff
@@ -45,14 +45,58 @@ func pathLogin(b *backend) *framework.Path {
...
+			logical.UpdateOperation:      b.pathLoginUpdate,
+			logical.ResolveRoleOperation: b.pathResolveRole,
+		},
+
+		HelpSynopsis:    pathLoginRoleSyn,
+		HelpDescription: pathLoginRoleDesc,
+	}
+}
+
+func pathLogin(b *backend) *framework.Path {
```
Comment (hghaf099) [inline path_login.go:57]: nit: do we need two path functions for this? I believe we can use a slice like `[]*framework.Path{..}` to add more paths.

File: path_login.go [hunk 2]
```diff
@@ -45,14 +45,58 @@ func pathLogin(b *backend) *framework.Path {
 			},
 		},
 		Callbacks: map[logical.Operation]framework.OperationFunc{
-			logical.UpdateOperation: b.pathLoginUpdate,
+			logical.UpdateOperation:      b.pathLoginUpdate,
+			logical.ResolveRoleOperation: b.pathResolveRole,
```
Comment (ccapurso) [inline path_login.go:49]: Given the comment that "core checks the /login endpoint for role resolution" in your description, will the `logical.ResolveRoleOperation` for `login/<role>` actually ever be hit due to role-based quotas?

Comment (ccapurso) [reply path_login.go:49]: I don't see any harm in it, was mostly wondering if it would have application outside role-based quotas.

Comment (VioletHynes) [reply path_login.go:49]: Right, makes sense. Yeah my thought was that no application is really targeted at this endpoint now but it seems like it has only potential benefits from being there

File: path_login.go [hunk 3]
```diff
@@ -244,10 +288,18 @@ func requestTargetToMethodURL(requestTarget []string, roleName string) (method s
...
 Authenticates to Vault using OCI credentials such as User Api Key, Instance Principal
 `
+
+const pathLoginSyn = `
+Determines the role that would be used for login from a valid OCI login request
+`
+
+const pathLoginDesc = `
+Determines the role that would be used for login from a valid OCI login request
+`
```
Comment (ccapurso) [inline path_login.go:305]: Should these actually be `pathLoginRoleSyn` and `pathLoginRoleDesc`?

Comment (ccapurso) [reply path_login.go:305]: Ah, whoops. That makes sense. I got it backwards. Thanks!

**Review by ccapurso — APPROVED** [review]
Looks good!

**Review by hghaf099 — APPROVED** [review]

**Review by tomhjp — APPROVED** [review]

**Pull request merged**